src/connection/DbConnection.js

```diff
@@ -167,6 +168,67 @@ module.exports = (
         });
     }
 
+    async function transactionQuery(sqls, multiValues = [], label) {
+        const outputLabel = label || DEFAULT_OUTPUT_LABEL;
+        const startToken = timers.start();
+
+        let connection;
+        try {
+            connection = await newConnection();
+        } catch (err) {
+            logger.error(`${outputLabel}.sql.connectionFailure`, err);
+            connection.destroy();
+            throw err;
+        }
+
+        // Promisified methods. Must be attached to keep 'this' context.
+        ['beginTransaction', 'query', 'rollback', 'commit'].forEach((method) => {
+            connection[`${method}Promise`] =
+                util.promisify(connection[method]);
+        });
+
+        try {
+            await connection.beginTransactionPromise();
+        } catch (err) {
+            logger.error(`${outputLabel}.sql.beginTransactionFailure`, err);
+            connection.destroy();
+            throw err;
+        }
+
+        const queryResults = [];
+
+        try {
```

Comment:
Use a `finally` block to clean up the connection like above. This one is especially important as `await connection.rollbackPromise();` could throw.